src/api.v2/routes/kubernetes.js

```diff
@@ -0,0 +1,33 @@
+const AWSXRay = require('aws-xray-sdk');
+const k8s = require('@kubernetes/client-node');
+const getLogger = require('../../utils/getLogger');
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+const logger = getLogger();
+
+module.exports = {
+  'kubernetes#event': async (req, res, next) => {
+    logger.log('received kubernetes event');
+    try {
+      const {
+        reason, message, type, involvedObject: { name, namespace },
+      } = req.body;
+      logger.log(`[${reason}] received ${type} kubernetes event: ${message} ${name} in ${namespace}`);
+
+      // remove only pods in your namespace and due to backoff errors
+      if ((namespace.match('^pipeline-.*') || namespace.match('^worker-.*'))
+       && reason === 'BackOff' && type !== 'Normal' && message.includes('restarting')) {
```

Comment:
Why does the condition need to be reason === BackOff and the message to include "restarting" is it not enough to delete the ones that are in BackOff?

Reply:
it turns out the `BackOff` is an umbrella term for many different issues. For example, when you are trying to download an image from dockerhub and you hit the rate limit, you'll get a backoff event (meaning that you are backing off and waiting before you retry the download again). After investigating this is the best way I managed to filter the restarts (documentation on kubernetes events is pretty bad and patchy, even checking the source code).